src/Object/Contact.php

```diff
@@ -21,29 +21,30 @@
 require_once 'boot.php';
 require_once 'include/text.php';
 
+/**
+ * @brief functions for interacting with a contact
+ */
 class Contact extends BaseObject
 {
 	/**
 	 * @brief Marks a contact for removal
 	 *
-	 * @param int $id
+	 * @param int $id contact id
 	 * @return null
 	 */
 	public static function remove($id)
 	{
 		// We want just to make sure that we don't delete our "self" contact
-		$r = q(
-			"SELECT `uid` FROM `contact` WHERE `id` = %d AND NOT `self` LIMIT 1", intval($id)
-		);
-		if (!DBM::is_result($r) || !intval($r[0]['uid'])) {
+		$condition = array('`id` = ? AND NOT `self`', $id);
+		$r = dba::select('contact', array('uid'), $condition, array('limit' => 1));
+
+		if (!DBM::is_result($r) || !intval($r['uid'])) {
 			return;
 		}
 
-		$archive = PConfig::get($r[0]['uid'], 'system', 'archive_removed_contacts');
+		$archive = PConfig::get($r['uid'], 'system', 'archive_removed_contacts');
 		if ($archive) {
-			q(
-				"UPDATE `contact` SET `archive` = 1, `network` = 'none', `writable` = 0 WHERE id = %d", intval($id)
-			);
+			dba::update('contact', array('archive' => 1, 'network' => 'none', 'writable' => 0), array('id' => $id));
```

Comment:
I, personally, like to use the boolean values to make it more readable, means:
```dba::update('contact', array('archive' => true, 'network' => 'none', 'writable' => false), array('id' => $id));```

It is tested that it works that way. But this is only my personal flavour.